content/projects/gsoc/2020/project-ideas/docker-registries-polling-plugin.adoc

```diff
@@ -0,0 +1,52 @@
+---
+layout: gsocprojectidea
+title: "Polling Docker Registries for Image Changes"
+goal: "Create a new Jenkins plugin to automate polling of image changes and security scans"
+category: Plugins
+year: 20
+status: published
+skills:
+- Java
+- Docker
+mentors:
+- "justinharringa"
+- "Andrey Falko"
+links:
+  gitter: jenkinsci/docker
+  draft: https://docs.google.com/document/d/1r_wOqtzmiIyiNWri6U3FKINWdnyHWEMF_lbSCa4jPiw
+---
+
+Keeping Docker images up-to-date is important to keep systems patched from known security vulnerabilities.
+Today, there isn’t a good way to subscribe to public docker images and update any of your own that depend on them.
+When depending on public images,
+Jenkins users are faced with several less-than-ideal choices:
+Build at a time cadence or manually trigger their Docker image builds.
+
+Today, a plugin:dockerhub-notification[Dockerhub Notification Plugin] exists which can help, but it requires a webhook.
+You can only use it if you own the images or the registry.
+The goal of this project is to be able to kick off your own Jenkins jobs when images that you don’t own change.
+For example, when link:https://hub.docker.com/_/debian/[debian:latest] has an update,
+we want to detect when it changes by polling. 
+The source registry should not matter.
+The ability to set default registry would be useful for seamless adoption by users running their own private registries.
+
+Mechanically, we need to keep track of the image digests for a given docker image tag.
+If a digest change is detected between polling periods,
+we need to start a build. If we are successful,
+a plugin can be installed by Jenkins users that can be used in both freestyle jobs and Jenkinsfiles to trigger the respective jobs.
+Alternative approaches are possible and welcome as well.
+
+Additional things that can be added is support for scanning target images for security vulnerabilities.
+Some organizations might only want to update their child images if there are changes related to security.
+The work to track image digests can be used to scan a set of target images that are created and provide alerts if vulnerabilities are found.
+The plugin:docker-traceability[Docker Traceability Plugin] can be reused to provide a lot of the heavy lifting.
+
+=== Quickstart
+
+* Setup a Jenkins instance, e.g. in Docker or Kubernetes.
+* Take a look at the link:https://github.com/jenkinsci/dockerhub-notification-plugin[jenkins Notification plugin] some of the concepts can be used in building the polling of the docker images.
+* Jenkins-x also achieves scanning target images for security vulnerabilities some part of it can be reused in the second phase of the project.
+
+=== Issues
+* link:https://issues.jenkins-ci.org/browse/JENKINS-59474?jql=project%20%3D%20JENKINS%20AND%20component%20%3D%20docker%20AND%20labels%20%3D%20newbie-friendly[Newbie-Friendly Issues].
```

Comment:
Link does not show the list of issues

Reply:
I did a search on jira with the component as docker and the label as newbie friendly and then just copied the link, do I need to do something different to get the list of issues

Reply:
Open that link in a "guest" or "icognito" window and you'll see what others will  see.

The query you've provided shows resolved and closed issues (confuse a new contributor, risk they work on something that is already fixed)

You might base your query on the [friendly issues dashboard](https://issues.jenkins-ci.org/secure/Dashboard.jspa?selectPageId=19342) like this [docker specific example](https://issues.jenkins-ci.org/issues/?jql=labels%20%3D%20newbie-friendly%20AND%20status%20not%20in%20(Closed%2C%20Done%2C%20Resolved%2C%20%22Fixed%20but%20Unreleased%22)%20AND%20component%20%3D%20docker%20AND%20project%20%3D%20JENKINS).

Reply:
Thanks @MarkEWaite I will correct the query :)

Reply:
Updated query on the document :)